test/confirmations_create_confirmation_request_unittest.cc

```diff
@@ -0,0 +1,167 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <string>
+
+#include "confirmations_client_mock.h"
+#include "brave/vendor/bat-native-confirmations/src/confirmations_impl.h"
+#include "brave/vendor/bat-native-confirmations/src/create_confirmation_request.h"
+#include "brave/vendor/bat-native-confirmations/src/security_helper.h"
+#include "brave/vendor/bat-native-confirmations/include/bat/confirmations/wallet_info.h"
+
+#include "testing/gtest/include/gtest/gtest.h"
+
+// npm run test -- brave_unit_tests --filter=Confirmations*
+
+namespace confirmations {
+
+class ConfirmationsCreateConfirmationRequestTest : public ::testing::Test {
+ protected:
+  MockConfirmationsClient* mock_confirmations_client_;
+  ConfirmationsImpl* confirmations_;
+
+  CreateConfirmationRequest* request_;
+
+  ConfirmationsCreateConfirmationRequestTest() :
+      mock_confirmations_client_(new MockConfirmationsClient()),
+      confirmations_(new ConfirmationsImpl(mock_confirmations_client_)),
+      request_(new CreateConfirmationRequest()) {
+    // You can do set-up work for each test here
+  }
+
+  ~ConfirmationsCreateConfirmationRequestTest() override {
+    // You can do clean-up work that doesn't throw exceptions here
+    delete request_;
+
+    delete confirmations_;
+    delete mock_confirmations_client_;
```

Comment:
Any particular reason not to use std::unique_ptr for these, so you can skip the explicit deletes?

Reply:
Used an old template for the unit tests, is this something we can fix later or prefer now? Thanks

Reply:
👍

Reply:
👍 I will fix this, thanks